src/tests/test_davidson.cc

```diff
@@ -131,6 +152,139 @@ BOOST_AUTO_TEST_CASE(davidson_matrix_free) {
   bool check_eigenvalues = lambda.isApprox(lambda_ref, 1E-6);
 
   BOOST_CHECK_EQUAL(check_eigenvalues, 1);
+  if (!check_eigenvalues) {
+    std::cout << "ref" << std::endl;
+    std::cout << es.eigenvalues().head(neigen).transpose() << std::endl;
+    std::cout << "result" << std::endl;
+    std::cout << DS.eigenvalues().transpose() << std::endl;
+  }
+}
+
+BOOST_AUTO_TEST_CASE(davidson_matrix_free_large) {
+
+  int size = 400;
+  int neigen = 10;
+
+  // Create Operator
+  TestOperator Aop;
+  Aop.set_size(size);
+
+  Logger log;
+  log.setReportLevel(logDEBUG);
+  DavidsonSolver DS(log);
+  DS.set_tolerance("normal");
+  DS.set_size_update("safe");
+  DS.solve(Aop, neigen);
+  std::cout << log << std::endl;
+  Eigen::MatrixXd A = Aop.get_full_matrix();
+  Eigen::SelfAdjointEigenSolver<Eigen::MatrixXd> es(A);
+
+  auto lambda = DS.eigenvalues();
+  auto lambda_ref = es.eigenvalues().head(neigen);
+  bool check_eigenvalues = lambda.isApprox(lambda_ref, 1E-6);
+
+  BOOST_CHECK_EQUAL(check_eigenvalues, 1);
+  if (!check_eigenvalues) {
+    std::cout << "ref" << std::endl;
+    std::cout << es.eigenvalues().head(neigen).transpose() << std::endl;
+    std::cout << "result" << std::endl;
+    std::cout << DS.eigenvalues().transpose() << std::endl;
+  }
+}
+
+class BlockOperator : public MatrixFreeOperator {
```

Comment:
I also have a class called BlockOperator for the A/B block of the BTDA case. That might conflict with that class ... so maybe we have to rename one or the other

Reply:
I renamed your test class as HermitianBlock